proto/rill/runtime/v1/queries.proto

```diff
@@ -605,3 +612,28 @@ message TableRowsRequest {
 message TableRowsResponse {
   repeated google.protobuf.Struct data = 1;
 }
+
+message DownloadLinkRequest {
+  int32 limit = 1;
+  DownloadFormat format = 2;
+  DownloadCompression compression = 3;
+  oneof request {
+    MetricsViewToplistRequest metrics_view_toplist_request = 4;
+    MetricsViewRowsRequest metrics_view_rows_request = 5;
+  }
+}
+
+enum DownloadFormat {
+  DOWNLOAD_FORMAT_UNSPECIFIED = 0; // assumes CSV
+  DOWNLOAD_FORMAT_CSV = 1;
+  DOWNLOAD_FORMAT_PARQUET = 2;
+}
+
+enum DownloadCompression {
+  DOWNLOAD_COMPRESSION_UNSPECIFIED = 0; // assumes no compression
+  DOWNLOAD_COMPRESSION_ZIP = 1;
+}
```

Comment:
To keep our backend simple, I would suggest not having compression be separate from format. For Parquet, compression is supposed to be applied at columnar level (i.e. inside the format, not at file level), so there we should just have an opinionated choice.

For CSV, we could have `DOWNLOAD_FORMAT_CSV` and `DOWNLOAD_FORMAT_CSV_GZIP` if necessary, though I think initially, if possible, we can just support CSV and instead apply gzip compression on the HTTP response (`Content-Encoding: gzip`) to minimize bandwidth